src/internal/_objectAssign.js

```diff
@@ -0,0 +1,24 @@
+var _has = require('./_has');
+
+// Based on https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
+module.exports = function _objectAssign(target) {
+  if (target === undefined || target === null) {
+    throw new TypeError('Cannot convert undefined or null to object');
+  }
+
+  var output = Object(target);
+  var idx = 1;
+  var length = arguments.length;
+  while (idx < length) {
+    var source = arguments[idx];
+    if (source !== undefined && source !== null) {
```

Comment:
I'll change that, but note that polyfills tend to explicitly favor this kind of verbosity.